src/VisualizationTests/HelixImageComparisonTests.cs

```diff
@@ -89,32 +92,72 @@ private void RenderCurrentViewAndCompare(string testName)
             var refbitmap = BitmapFromSource(refimage);
             var newImage = BitmapFromSource(bitmapsource);
 
+#if SAVEDEBUGIMAGES
+            var debugPath = GenerateTestDataPathFromTest(testName,true);
+            SaveBitMapSourceAsPNG(debugPath, bitmapsource);
+#endif
+
             compareImageColors(refbitmap, newImage);
         }
-
-        //TODO consider something like:
-        //(diferentPixelsCount / (mainImage.width* mainImage.height))*100
-        //for a percent image diff.
         private static void compareImageColors(Bitmap image1,Bitmap image2)
         {
             Assert.AreEqual(image1.Width, image2.Width);
             Assert.AreEqual(image1.Height, image2.Height);
             Assert.AreEqual(image1.PixelFormat, image2.PixelFormat);
 
+            //x,y,expected,result
+            var differences = new List<Tuple<int, int, Color, Color>>();
+
             for (var x = 0; x < image1.Width; x++)
             {
                 for (var y = 0; y < image1.Height; y++)
                 {
                     var expectedCol = image1.GetPixel(x, y);
                     var otherCol = image2.GetPixel(x, y);
-                    Assert.AreEqual(expectedCol, otherCol,$"pixel {x}:{y} was not as expected");
+                    if((ColorDistance(expectedCol,otherCol) > 128))
+                    {
+                        differences.Add(Tuple.Create(x, y, expectedCol, otherCol));
+                    // this can be painfully slow, but is useful during debug - uncomment for more info.
+                    // Console.WriteLine($"{expectedCol}, {otherCol}, pixel {x}:{y} was not in expected range");
+                    }
                 }
             }
+            var diff = CalculatePercentDiff(image1, differences);
+            Console.WriteLine($"% difference by out of range pixels was {(diff * 100).ToString("N" + 3)}%");
+            Assert.LessOrEqual(diff, .01);
+
+        }
+
+        /// <summary>
+        /// Euclidean distance between colors.
+        /// Does not use the alpha channel.
```

Comment:
Why is the alpha channel not used?

Reply:
theres no alpha channel in a full color image with no transparency. These are renders not textures or masks.

Reply:
What about when we are comparing images in isolation mode or frozen geometry?

Reply:
nope, even then, the final output color is just red,green,blue, the alpha channel is still 0. These are 24bit images anyway, there isn't even an alpha channel.

Reply:
So would the image comparison between an isolated geometry and non-isolated geometry return true (same image)?

Reply:
Is this the reason why @QilongTang couldn't write tests to check if isolated geometry was being rendered as expected?

Reply:
no!
I'm not doing a good job explaining this :)

When we set the alpha channel to control transparency that has no real meaning in terms of color - how alpha channel changes the output of the frag shader is a decision the renderer/shader makes when it determines the final color of the rendered output.

So when we generate a bitmap the alpha channel has already been incorporated into the final RGB color - you can think of this like premultiplying or something - the color values have been modified by the alpha channel in some way to create a color blended with the other colors that were behind those pixels.

The resulting alpha channel is meaningless for the final image here because we're not creating a mask or other texture where we consider it to have some meaning - we only care about the final color.

Reply:
@aparajit-pratap I think so, alpha value did not indicate anything for the isolated geometry case in the old `master` branch.

Reply:
thats unrelated - in your test you were not looking at the color values of an image, you were looking at the color values of individual verts which were not used for computing the final color.

Reply:
@mjkkirschner OK, let me read your comments carefully again!

Reply:
I know it's a bit confusing - but you can see in the shader:

```
//if isolated - alpha should always be low - 
//overriding other alpha values (except if selected)
if(isIsolated && !isSelected)
{
	I.a = .1f;
}
```

that if the isolate mode is active we set the alpha of the returned color (returned from the shader)

Because the shader is using AlphaBlending - the renderer uses this alpha to compute a final color.

I hope that makes sense. Please let me know if not, I can find some better written docs on the subject.

Reply:
> The resulting alpha channel is meaningless for the final image here because we're not creating a mask or other texture where we consider it to have some meaning - we only care about the final color.

Ah, so it's something like homogeneous coordinates :)

Reply:
> thats unrelated - in your test you were not looking at the color values of an image, you were looking at the color values of individual verts which were not used for computing the final color.

Yup, sorry for confusing the two; my bad.

Reply:
> So when we generate a bitmap the alpha channel has already been incorporated into the final RGB color - you can think of this like premultiplying or something - the color values have been modified by the alpha channel in some way to create a color blended with the other colors that were located at the same pixels.

Ah, so it's something like homogeneous coordinates :)

Reply:
Make sense! @mjkkirschner